In [1]:
%config Completer.use_jedi = False
%load_ext autoreload

# Imports and Functions

In [44]:
import pandas as pd
import csv
import numpy as np

# Import CSVs

In [193]:
df_publishers = pd.read_csv("../../data/ingest/game_data/first_publishers.csv").rename(columns={"id":"bgg_id"})
print(df_publishers.shape)
df_publishers.head()

(175041, 2)


,bgg_id,publisher
0,1,Hans im Glück
1,2,E. S. Lowe Company Inc.
2,3,Fantasy Flight Games
3,4,KOSMOS
4,5,3M


In [194]:
df_bgg = pd.read_csv("../../data/ingest/ranks/boardgame_ranks_20260319.csv", sep="|", escapechar="\\").rename(columns={"id":"bgg_id", "name":"bgg_name"})
print(df_bgg.shape)
df_bgg = df_bgg.merge(df_publishers, on="bgg_id", how="left")
df_bgg.head()

(175041, 17)


,bgg_id,bgg_name,yearpublished,rank,bayesaverage,average,usersrated,is_expansion,abstracts_rank,cgs_rank,childrensgames_rank,familygames_rank,partygames_rank,strategygames_rank,thematic_rank,wargames_rank,queried_at_utc,publisher
0,224517,Brass: Birmingham,2018,1,8.39496,8.56730,57338,0,NaN,NaN,NaN,NaN,NaN,1.0,NaN,NaN,2026-03-19T20:38:04,Roxley
1,342942,Ark Nova,2021,2,8.35511,8.54201,59804,0,NaN,NaN,NaN,NaN,NaN,2.0,NaN,NaN,2026-03-19T20:38:04,Feuerland Spiele
2,161936,Pandemic Legacy: Season 1,2015,3,8.34782,8.50436,57147,0,NaN,NaN,NaN,NaN,NaN,3.0,1.0,NaN,2026-03-19T20:38:04,Z-Man Games
3,174430,Gloomhaven,2017,4,8.30193,8.54303,66820,0,NaN,NaN,NaN,NaN,NaN,5.0,2.0,NaN,2026-03-19T20:38:04,Cephalofair Games
4,397598,Dune: Imperium – Uprising,2023,5,8.22545,8.70472,17374,0,NaN,NaN,NaN,NaN,NaN,4.0,NaN,NaN,2026-03-19T20:38:04,Dire Wolf


In [136]:
df_paxtt = pd.read_csv("../../data/pax/title_corrections_raw.csv", sep="|", encoding="cp1252")
# for col in ["bgg_id", "game_id"]:
#     df_paxtt[col] = df_paxtt[col].astype(pd.Int64Dtype())
# df_paxtt = df_paxtt.drop(columns=["corrected_title", "corrected_bgg_id"])
df_paxtt["bgg_id"] = df_paxtt["bgg_id"].astype(pd.Int64Dtype())
df_paxtt.loc[df_paxtt["bgg_id"]==0, "bgg_id"] = np.nan
print(df_paxtt.shape)
df_paxtt.head()

(11418, 7)


,title,title_id,game_id,game_barcode,publisher_name,bgg_id,game_year
0,"""A Matter of Honor, Sire""",1148,2680,TTL1019O,AEG - DO NOT USE,<NA>,2014
1,#artbox,5025,10440,TTL06494I,Lavka Games,<NA>,2024
2,1 A.M. Jailbreak,5189,10805,25EAST0418,Saashi & Saashi,<NA>,2025
3,1 A.M. Jailbreak,5189,10806,25EAST0414,Saashi & Saashi,<NA>,2025
4,1 A.M. Jailbreak,5189,10807,25EAST0410,Saashi & Saashi,<NA>,2025


# Merge the data

## Manually fix bad bgg_ids

In [208]:
df_paxtt.loc[df_paxtt["bgg_id"]==89862, "bgg_id"] = 55253
df_paxtt.loc[df_paxtt["bgg_id"]==234497, "bgg_id"] = 242409
df_paxtt.loc[df_paxtt["bgg_id"]==268191, "bgg_id"] = 244271
df_paxtt.loc[df_paxtt["bgg_id"]==224965, "bgg_id"] = 72376
df_paxtt.loc[df_paxtt["bgg_id"]==63993, "bgg_id"] = 2290
df_paxtt.loc[df_paxtt["bgg_id"]==282031, "bgg_id"] = 364358

In [209]:
df_paxtt.loc[df_paxtt["title_id"]==387, "bgg_id"] = 224
df_paxtt.loc[df_paxtt["title_id"]==2443, "bgg_id"] = 224
df_paxtt.loc[df_paxtt["title_id"]==423, "bgg_id"] = 34010
df_paxtt.loc[df_paxtt["title_id"]==3830, "bgg_id"] = 324246
df_paxtt.loc[df_paxtt["title_id"]==1197, "bgg_id"] = 20437
df_paxtt.loc[df_paxtt["title_id"]==3601, "bgg_id"] = 20437

In [221]:
df_paxtt.loc[df_paxtt["title_id"]==1544, "bgg_id"] = 183231
df_paxtt.loc[df_paxtt["title_id"]==4311, "bgg_id"] = 357563
df_paxtt.loc[df_paxtt["title_id"]==4341, "bgg_id"] = 370391
df_paxtt.loc[df_paxtt["title_id"]==4146, "bgg_id"] = 359402
df_paxtt.loc[df_paxtt["title_id"]==992, "bgg_id"] = 12248
df_paxtt.loc[df_paxtt["title_id"]==3360, "bgg_id"] = 317983
df_paxtt.loc[df_paxtt["title_id"]==3858, "bgg_id"] = 317983
df_paxtt.loc[df_paxtt["title_id"]==5019, "bgg_id"] = 420214
df_paxtt.loc[df_paxtt["title_id"]==1912, "bgg_id"] = 207208
df_paxtt.loc[df_paxtt["title_id"]==5208, "bgg_id"] = 419454
df_paxtt.loc[df_paxtt["title_id"]==49, "bgg_id"] = 69789
df_paxtt.loc[df_paxtt["title_id"]==4649, "bgg_id"] = 348554
df_paxtt.loc[df_paxtt["title_id"]==2738, "bgg_id"] = 446
df_paxtt.loc[df_paxtt["title_id"]==60, "bgg_id"] = 10093
df_paxtt.loc[df_paxtt["title_id"]==5164, "bgg_id"] = 418354
df_paxtt.loc[df_paxtt["title_id"]==5038, "bgg_id"] = 418354
df_paxtt.loc[df_paxtt["title_id"]==66, "bgg_id"] = 43231
df_paxtt.loc[df_paxtt["title_id"]==73, "bgg_id"] = 551
df_paxtt.loc[df_paxtt["title_id"]==75, "bgg_id"] = 760
df_paxtt.loc[df_paxtt["title_id"]==5280, "bgg_id"] = 402674
df_paxtt.loc[df_paxtt["title_id"]==2023, "bgg_id"] = 2425
df_paxtt.loc[df_paxtt["title_id"]==5532, "bgg_id"] = 446748
df_paxtt.loc[df_paxtt["title_id"]==5192, "bgg_id"] = 424730
df_paxtt.loc[df_paxtt["title_id"]==5184, "bgg_id"] = 421310
df_paxtt.loc[df_paxtt["title_id"]==5216, "bgg_id"] = 420805
df_paxtt.loc[df_paxtt["title_id"]==4301, "bgg_id"] = 367891
df_paxtt.loc[df_paxtt["title_id"]==2146, "bgg_id"] = 231824
df_paxtt.loc[df_paxtt["title_id"]==4506, "bgg_id"] = 366994
df_paxtt.loc[df_paxtt["title_id"]==174, "bgg_id"] = 1294
df_paxtt.loc[df_paxtt["title_id"]==3693, "bgg_id"] = 329450
df_paxtt.loc[df_paxtt["title_id"]==1205, "bgg_id"] = 131121
df_paxtt.loc[df_paxtt["title_id"]==5403, "bgg_id"] = 395623
df_paxtt.loc[df_paxtt["title_id"]==2080, "bgg_id"] = 200058
df_paxtt.loc[df_paxtt["title_id"]==2503, "bgg_id"] = 316093
df_paxtt.loc[df_paxtt["title_id"]==3673, "bgg_id"] = 316093
df_paxtt.loc[df_paxtt["title_id"]==5122, "bgg_id"] = 400265
df_paxtt.loc[df_paxtt["title_id"]==4737, "bgg_id"] = 3076
df_paxtt.loc[df_paxtt["title_id"]==755, "bgg_id"] = 37196
df_paxtt.loc[df_paxtt["title_id"]==4422, "bgg_id"] = 147020
df_paxtt.loc[df_paxtt["title_id"]==3966, "bgg_id"] = 84876

In [ ]:
df_paxtt.loc[df_paxtt["title_id"]==5155, "bgg_id"] = 552
df_paxtt.loc[df_paxtt["title_id"]==3879, "bgg_id"] = 290236
df_paxtt.loc[df_paxtt["title_id"]==3269, "bgg_id"] = 170583
df_paxtt.loc[df_paxtt["title_id"]==148, "bgg_id"] = 65521
df_paxtt.loc[df_paxtt["title_id"]==167, "bgg_id"] = 478
df_paxtt.loc[df_paxtt["title_id"]==1857, "bgg_id"] = 478
df_paxtt.loc[df_paxtt["title_id"]==5497, "bgg_id"] = 205398
df_paxtt.loc[df_paxtt["title_id"]==2715, "bgg_id"] = 1294
df_paxtt.loc[df_paxtt["title_id"]==2720, "bgg_id"] = 1294
df_paxtt.loc[df_paxtt["title_id"]==1306, "bgg_id"] = 145639
df_paxtt.loc[df_paxtt["title_id"]==181, "bgg_id"] = 24742
df_paxtt.loc[df_paxtt["title_id"]==5135, "bgg_id"] = 394887
df_paxtt.loc[df_paxtt["title_id"]==1925, "bgg_id"] = 176306
df_paxtt.loc[df_paxtt["title_id"]==5239, "bgg_id"] = 424435
df_paxtt.loc[df_paxtt["title_id"]==5483, "bgg_id"] = 396895
df_paxtt.loc[df_paxtt["title_id"]==194, "bgg_id"] = 39463
df_paxtt.loc[df_paxtt["title_id"]==1093, "bgg_id"] = 121414
df_paxtt.loc[df_paxtt["title_id"]==1503, "bgg_id"] = 131357
df_paxtt.loc[df_paxtt["title_id"]==3857, "bgg_id"] = 317981
df_paxtt.loc[df_paxtt["title_id"]==2148, "bgg_id"] = 225694
df_paxtt.loc[df_paxtt["title_id"]==4582, "bgg_id"] = 376747
df_paxtt.loc[df_paxtt["title_id"]==4457, "bgg_id"] = 376747
df_paxtt.loc[df_paxtt["title_id"]==3803, "bgg_id"] = 317457
df_paxtt.loc[df_paxtt["title_id"]==241, "bgg_id"] = 6137
df_paxtt.loc[df_paxtt["title_id"]==243, "bgg_id"] = 121
df_paxtt.loc[df_paxtt["title_id"]==5041, "bgg_id"] = 181329
df_paxtt.loc[df_paxtt["title_id"]==5205, "bgg_id"] = 181329
df_paxtt.loc[df_paxtt["title_id"]==269, "bgg_id"] = 187
df_paxtt.loc[df_paxtt["title_id"]==269, "bgg_id"] = 72125
df_paxtt.loc[df_paxtt["title_id"]==283, "bgg_id"] = 33160
df_paxtt.loc[df_paxtt["title_id"]==3110, "bgg_id"] = 131121
df_paxtt.loc[df_paxtt["title_id"]==3588, "bgg_id"] = 329450
df_paxtt.loc[df_paxtt["title_id"]==5010, "bgg_id"] = 379
df_paxtt.loc[df_paxtt["title_id"]==4985, "bgg_id"] = 385761
df_paxtt.loc[df_paxtt["title_id"]==5042, "bgg_id"] = 439209
df_paxtt.loc[df_paxtt["title_id"]==300, "bgg_id"] = 40628
df_paxtt.loc[df_paxtt["title_id"]==4988, "bgg_id"] = 40628
df_paxtt.loc[df_paxtt["title_id"]==5298, "bgg_id"] = 419195
df_paxtt.loc[df_paxtt["title_id"]==5043, "bgg_id"] = 374024
df_paxtt.loc[df_paxtt["title_id"]==4329, "bgg_id"] = 376218
df_paxtt.loc[df_paxtt["title_id"]==5302, "bgg_id"] = 417518
df_paxtt.loc[df_paxtt["title_id"]==3368, "bgg_id"] = 275969

## Merge on bgg_id

In [222]:
df_m1 = df_paxtt.merge(df_bgg[["bgg_name", "bgg_id"]], on="bgg_id", how="inner")
df_m1.shape

(3947, 8)

In [223]:
df_m1.loc[df_m1["bgg_name"].notna()].head(10)

,title,title_id,game_id,game_barcode,publisher_name,bgg_id,game_year,bgg_name
0,12 Rivers,4341,9137,TTL32241I,Dofinn,370391,2022,12 Rivers
1,1920 Wall Street,1989,4259,TTL04221I,Looping Games,226255,2017,1920 Wall Street
2,1942 USS Yorktown,3378,7133,TTL06189I,Looping Games,279307,2019,1942 USS Yorktown
3,1942 USS Yorktown,3378,7134,TTL06193I,Looping Games,279307,2019,1942 USS Yorktown
4,1942 USS Yorktown,3378,7135,TTL06197I,Looping Games,279307,2019,1942 USS Yorktown
5,1987 Channel Tunnel,3379,7136,TTL06201I,Looping Games,279306,2019,1987 Channel Tunnel
6,1987 Channel Tunnel,3379,7137,TTL06209I,Looping Games,279306,2019,1987 Channel Tunnel
7,1987 Channel Tunnel,3379,7138,TTL06205I,Looping Games,279306,2019,1987 Channel Tunnel
8,3 Laws of Robotics,3324,6992,TTL06239I,Floodgate Games,271447,2019,3 Laws of Robotics
9,5-Minute Marvel,3191,6726,TTL09200I,Spin Master,253618,2019,5-Minute Marvel


# Merge on exact name, year, and publisher

In [236]:
df_m2 = df_paxtt.loc[~(df_paxtt["bgg_id"].isin(df_m1["bgg_id"])),['title', 'title_id', 'game_barcode', 'publisher_name', "game_year"]].drop_duplicates()
df_m2["title_lower"] = df_m2["title"].str.lower()
df_bgg["title_lower"] = df_bgg["bgg_name"].str.lower()
df_m2["publisher_lower"] = df_m2["publisher_name"].str.lower()
df_bgg["publisher_lower"] = df_bgg["publisher"].str.lower()
df_m2 = df_m2.merge(df_bgg[["bgg_id", "title_lower", "usersrated", "yearpublished", "publisher_lower"]], on=["title_lower", "publisher_lower"], how="inner")
df_m2 = df_m2.loc[df_m2["game_year"]+1>= df_m2["yearpublished"]]
df_m2 = df_m2.drop(columns=["title_lower", "publisher_lower"])
df_m2.shape

(2661, 8)

In [225]:
duplicated_bgg_id = df_m2[["title", "bgg_id"]].drop_duplicates()
duplicated_bgg_id = duplicated_bgg_id.loc[duplicated_bgg_id["bgg_id"].duplicated(), "bgg_id"].tolist()
len(duplicated_bgg_id)

23

In [215]:
df_m2.loc[df_m2["bgg_id"].isin(duplicated_bgg_id[5:10])]

,title,title_id,game_barcode,publisher_name,game_year,bgg_id,usersrated,yearpublished
568,Descent: Journeys in the Dark,216,NaN,Fantasy Flight Games,2014,17226,11874,2005
569,Descent: Journeys in the Dark,216,TTL100MT,Fantasy Flight Games,2014,17226,11874,2005
570,Descent: Journeys in the dark,217,TTL1011O,Fantasy Flight Games,2014,17226,11874,2005
602,Dogfight!: Rule The Skies in 20 Minutes!,4222,TTL16715I,Psc Games,2022,369154,390,2023
603,Dogfight!: Rule The Skies in 20 Minutes!,4222,TTL32344I,Psc Games,2023,369154,390,2023
604,Dogfight!: Rule The Skies in 20 Minutes!,4222,TTL32340I,Psc Games,2023,369154,390,2023
605,Dogfight!: Rule The Skies in 20 Minutes!,4222,TTL09972I,Psc Games,2023,369154,390,2023
606,Dogfight!: Rule The Skies in 20 Minutes!,4222,TTL09968I,Psc Games,2023,369154,390,2023
607,Dogfight!: Rule The Skies In 20 Minutes!,4761,TTL30343I,Psc Games,2024,369154,390,2023
939,Gift Of Tulips,4220,TTL16651I,Weird Giraffe Games,2022,286788,780,2022


In [226]:
duplicated_title = df_m2[["title", "bgg_id"]].drop_duplicates()
duplicated_title = duplicated_title.loc[duplicated_title["title"].duplicated(), "title"].tolist()
len(duplicated_title)

0

In [ ]:




df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000


In [227]:
# df_m2.loc[df_m2["title"].isin(duplicated_title[10:15])]

# Merge on name exact

In [256]:
df_m3 = df_paxtt.loc[~(df_paxtt["title_id"].isin(df_m1["title_id"].tolist() + df_m2["title_id"].tolist())),['title', 'title_id', 'game_barcode', 'publisher_name', "game_year"]].drop_duplicates()
df_m3["title_lower"] = df_m3["title"].str.lower()
df_bgg["title_lower"] = df_bgg["bgg_name"].str.lower()
df_m3 = df_m3.merge(df_bgg[["bgg_name", "bgg_id", "title_lower", "usersrated", "yearpublished", "publisher"]], on="title_lower", how="inner")
df_m3 = df_m3.loc[df_m3["game_year"]+1>= df_m3["yearpublished"]]
df_m3.shape

(3531, 11)

In [257]:
df_m3.head()

,title,title_id,game_barcode,publisher_name,game_year,title_lower,bgg_name,bgg_id,usersrated,yearpublished,publisher
0,"""A Matter of Honor, Sire""",1148,TTL1019O,AEG - DO NOT USE,2014,"""a matter of honor, sire""","""A Matter of Honor, Sire""",153128,22,2013,Against the Odds
1,10 days in asia,1,TTL100RV,Out of the Box,2014,10 days in asia,10 Days in Asia,22398,1204,2007,Out of the Box Publishing
2,1313 Dead End Drive,2053,TTL04115I,Parker Brothers,2017,1313 dead end drive,1313 Dead End Drive,4953,981,2002,Milton Bradley
3,1775: Rebellion,1946,TTL03505I,Academy Games,2017,1775: rebellion,1775: Rebellion,128996,4468,2013,"Academy Games, Inc."
4,1775: Rebellion,1946,TTL03505I,Academy Games,2018,1775: rebellion,1775: Rebellion,128996,4468,2013,"Academy Games, Inc."


In [258]:
duplicated_bgg_id = df_m3[["title", "bgg_id"]].drop_duplicates()
duplicated_bgg_id = duplicated_bgg_id.loc[duplicated_bgg_id["bgg_id"].duplicated(), "bgg_id"].tolist()
len(duplicated_bgg_id)

22

In [259]:
duplicated_title = df_m3[["title", "bgg_id"]].drop_duplicates()
duplicated_title = duplicated_title.loc[duplicated_title["title"].duplicated(), "title"].tolist()
len(duplicated_title)

244

In [261]:
df_m3.loc[df_m3["bgg_id"].isin(duplicated_bgg_id[0:10])]

,title,title_id,game_barcode,publisher_name,game_year,title_lower,bgg_name,bgg_id,usersrated,yearpublished,publisher
166,Ascension: Rise of Vigil,52,TTL02074I,Stoneblade Entertainment,2014,ascension: rise of vigil,Ascension: Rise of Vigil,138233,1823,2013,asmodee
167,Ascension: Rise of Vigil,52,TTL02078I,Stoneblade Entertainment,2014,ascension: rise of vigil,Ascension: Rise of Vigil,138233,1823,2013,asmodee
168,Ascension: Rise of Vigil,52,TTL101B7,Stoneblade Entertainment,2014,ascension: rise of vigil,Ascension: Rise of Vigil,138233,1823,2013,asmodee
169,Ascension: Rise Of Vigil,4211,TTL16658I,Capstone Games,2022,ascension: rise of vigil,Ascension: Rise of Vigil,138233,1823,2013,asmodee
170,Ascension: Storm of Souls,53,TTL1010V,Stoneblade Entertainment,2014,ascension: storm of souls,Ascension: Storm of Souls,108784,3704,2011,Stone Blade Entertainment
171,Ascension: Storm of Souls,53,TTL1012G,Stoneblade Entertainment,2014,ascension: storm of souls,Ascension: Storm of Souls,108784,3704,2011,Stone Blade Entertainment
172,Ascension: Storm of Souls,53,TTL02178I,Stoneblade Entertainment,2014,ascension: storm of souls,Ascension: Storm of Souls,108784,3704,2011,Stone Blade Entertainment
173,Ascension: Storm Of Souls,1747,TTL02178I,Gary Games,2017,ascension: storm of souls,Ascension: Storm of Souls,108784,3704,2011,Stone Blade Entertainment
179,Attack the Darkness,1100,TTL02179I,Protean Games,2014,attack the darkness,Attack the Darkness,131496,46,2013,(Self-Published)
180,Attack the Darkness,1100,TTL02172I,Protean Games,2014,attack the darkness,Attack the Darkness,131496,46,2013,(Self-Published)


In [ ]:

df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000
df_paxtt.loc[df_paxtt["title_id"]==0000, "bgg_id"] = 0000


In [271]:
df_m3.loc[df_m3["title"].isin(duplicated_title[60:70])]

,title,title_id,game_barcode,publisher_name,game_year,title_lower,bgg_name,bgg_id,usersrated,yearpublished,publisher
1130,Food Fight,309,TTL100MN,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,99097,746,2011,Cryptozoic Entertainment
1131,Food Fight,309,TTL100MN,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,2193,70,1980,Tactical Studies Rules (TSR)
1132,Food Fight,309,TTL100MN,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,26781,0,2006,Avalon Games
1134,Food Fight,309,TTL100MO,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,99097,746,2011,Cryptozoic Entertainment
1135,Food Fight,309,TTL100MO,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,2193,70,1980,Tactical Studies Rules (TSR)
1136,Food Fight,309,TTL100MO,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,26781,0,2006,Avalon Games
1138,Food Fight,309,TTL100QK,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,99097,746,2011,Cryptozoic Entertainment
1139,Food Fight,309,TTL100QK,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,2193,70,1980,Tactical Studies Rules (TSR)
1140,Food Fight,309,TTL100QK,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,26781,0,2006,Avalon Games
1142,Food Fight,309,TTL100QO,Cryptozoic- DO NOT USE,2014,food fight,Food Fight,99097,746,2011,Cryptozoic Entertainment


Look for games with the same name but different bgg ids

In [99]:
len(duplicated_bgg_id)

62

In [89]:
df_m2[["bgg_id"]].drop_duplicates().shape

(3318, 1)

In [68]:
for k,v in df_m1.loc[(df_m1["bgg_name"].str.lower()!=df_m1["title"].str.lower())].iterrows():
    print(v.title, "|", v.bgg_name, "|",v.bgg_id, "|", v.game_id)

6 nimmt! | Beat the Heat | 432 | 2951
A Game of Thrones: Catan  Brotherhood of the Watch | A Game of Thrones: Catan – Brotherhood of the Watch | 229218 | 4344
A Game of Thrones: Catan  Brotherhood of the Watch | A Game of Thrones: Catan – Brotherhood of the Watch | 229218 | 4801
A Game of Thrones: Catan  Brotherhood of the Watch | A Game of Thrones: Catan – Brotherhood of the Watch | 229218 | 4780
Arkham Noir: Case #1  The Witch Cult Murders | Arkham Noir: Case #1 – The Witch Cult Murders | 225729 | 6331
Arkham Noir: Case #1  The Witch Cult Murders | Arkham Noir: Case #1 – The Witch Cult Murders | 225729 | 6332
Arkham Noir: Case #1  The Witch Cult Murders | Arkham Noir: Case #1 – The Witch Cult Murders | 225729 | 6333
Atelier: The Painter's Studio | Atelier | 311711 | 6932
Atelier: The Painter's Studio | Atelier | 311711 | 6963
Bloom in Blizzard | 花と吹雪 (Bloom in Blizzard) | 261431 | 6886
Cat Cafe | Cat Café | 279135 | 7319
Cat Cafe | Cat Café | 279135 | 7320
Cat Cafe | Cat Café | 27913

Inspect list game_ids (Do not alter their names)
- 2951

In [55]:
df_m1.loc[(df_m1["name"].str.lower()!=df_m1["title"].str.lower())].head(50)

,title,title_id,game_id,game_barcode,publisher_name,bgg_id,name
13,6 nimmt!,1291,2951,TTL02693I,Mayfair Games,432,Beat the Heat
35,A Game of Thrones: Catan Brotherhood of the W...,2061,4344,TTL05177I,Fantasy Flight Games,229218,A Game of Thrones: Catan – Brotherhood of the ...
36,A Game of Thrones: Catan Brotherhood of the W...,2061,4801,TTL04365I,Fantasy Flight Games,229218,A Game of Thrones: Catan – Brotherhood of the ...
37,A Game of Thrones: Catan Brotherhood of the W...,2307,4780,TTL04321I,Fantasy Flight Games,229218,A Game of Thrones: Catan – Brotherhood of the ...
165,Arkham Noir: Case #1 The Witch Cult Murders,3026,6331,TTL03556I,Ludonova,225729,Arkham Noir: Case #1 – The Witch Cult Murders
166,Arkham Noir: Case #1 The Witch Cult Murders,3026,6332,TTL03560I,Ludonova,225729,Arkham Noir: Case #1 – The Witch Cult Murders
167,Arkham Noir: Case #1 The Witch Cult Murders,3026,6333,TTL03599I,Ludonova,225729,Arkham Noir: Case #1 – The Witch Cult Murders
172,Atelier: The Painter's Studio,3295,6932,TTL06016I,AEG,311711,Atelier
173,Atelier: The Painter's Studio,3295,6963,TTL06145I,AEG,311711,Atelier
304,Bloom in Blizzard,3275,6886,PAX86488C,Akihiroitoo,261431,花と吹雪 (Bloom in Blizzard)


In [ ]:
df_missing_1 = df_sheet.loc[~(df_sheet["name"].isin(df_m1["name"]))].copy()
df_missing_1.shape